File: old/actions/install_base_tools.py
"""
Install base tools action
"""
from .base import Action
from cli import Apt

class InstallBaseToolsAction(Action):
    """Install minimal base tools"""
    description = "base tools installation"

    def execute(self) -> bool:
        """Install base tools"""
        install_cmd = Apt().install(["ca-certificates", "curl"])
        output = self.apt_service.execute(install_cmd)
        if output is None:
            self.logger.error("Failed to install base tools")
            return False
        return True

File: src/actions/install_base_tools.go
package actions

import (
	"enva/libs"
	"enva/services"
)

// InstallBaseToolsAction installs minimal base tools
type InstallBaseToolsAction struct {
	*BaseAction
}

func NewInstallBaseToolsAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &InstallBaseToolsAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *InstallBaseToolsAction) Description() string {
	return "base tools installation"
}

func (a *InstallBaseToolsAction) Execute() bool {
	output, exitCode := a.APTService.Install([]string{"ca-certificates", "curl"})
	if exitCode == nil || *exitCode != 0 {
		libs.GetLogger("install_base_tools").Printf("Failed to install base tools: %s", output)
		return false
	}
	return true
}

